### PR TITLE
stm32/sdcard: Ability to support boards with no SD card detect PIN

### PR DESCRIPTION
**Pull request by abelykh0**
Ability to support boards with no SD card detect PIN, such as https://github.com/mcauser/BLACK_F407VE.

### CODE REVIEW

**Comment by mcauser** [comment]
Also an issue on other MCUDev boards. MicroSD SW1 hard wired to GND:
* https://github.com/mcauser/BLACK_F407ZE
* https://github.com/mcauser/BLACK_F407ZG
* https://github.com/mcauser/MCUDEV_DEVEBOX_F407VET6
* https://github.com/mcauser/MCUDEV_DEVEBOX_F407VGT6

The VCC-GND boards are good. They have detect pins connected to SW1:
https://github.com/mcauser/VCC_GND_F407VE - PA8
https://github.com/mcauser/VCC_GND_F407ZG - PF10

File: ports/stm32/sdcard.c [hunk 1]
```diff
@@ -175,7 +175,9 @@ void sdcard_init(void) {
 
     // configure the SD card detect pin
     // we do this here so we can detect if the SD card is inserted before powering it on
-    mp_hal_pin_config(MICROPY_HW_SDCARD_DETECT_PIN, MP_HAL_PIN_MODE_INPUT, MICROPY_HW_SDCARD_DETECT_PULL, 0);
+	#if defined(MICROPY_HW_SDCARD_DETECT_PRESENT)
```
Comment (dpgeorge) [inline ports/stm32/sdcard.c]: I would suggest to use `#if defined(MICROPY_HW_SDCARD_DETECT_PIN)` instead.

Comment (abelykh0) [reply ports/stm32/sdcard.c]: OK, makes perfect sense. Updated.

**Comment by dpgeorge** [comment]
Thanks for updating, merged in 305f537bf940fbbe3474cd1e0938586426d25b48

**Pull request closed without merging**